app/models/KafkaMessageReader.scala

```diff
@@ -0,0 +1,109 @@
+package models
+
+import java.io.{ByteArrayInputStream, InputStreamReader}
+import java.util.zip.{GZIPInputStream, ZipException}
+
+import com.google.common.io.CharStreams
+import kafka.api.{FetchRequestBuilder, PartitionMetadata, TopicMetadataRequest}
+import kafka.consumer.SimpleConsumer
+import play.api.Play.current
+import play.api.{Logger, Play}
+
+/**
+  * Contains code to read messages from kafka.
+  */
+object KafkaMessageReader {
+
+  private val Kafkas = Play.configuration.getString("capillary.kafkas").getOrElse("127.0.0.1:9092")
+  private val KafkaHosts = Kafkas.split(",")
+
+  private val Timeout = 100000
+  private val BufferSize = 10000000
+  private val ClientId = "Capillary"
+
+  def getMessageContent(topic: String, partition: Int, offset: Long): String = {
+    findLeader(KafkaHosts, topic, partition) match {
+      case Some(pm) => readMessage(pm, topic, offset)
+      case None => s"Could not find any leader for partition: $partition"
+    }
+  }
+
+  private def findLeader(kafkaHosts: Array[String], topic: String, partition: Int): Option[PartitionMetadata] = {
+    var simpleConsumer: Option[SimpleConsumer] = None
+    try {
+      for (kafkaHost <- kafkaHosts) {
+        val hostAndPort = kafkaHost.split(":")
+        val host = hostAndPort(0)
+        val port = hostAndPort(1).toInt
+        simpleConsumer = Some(new SimpleConsumer(host, port, Timeout, BufferSize, ClientId))
```

Comment:
Maybe I'm missing something, but doesn't this leak every consumer instance it creates up until it finds one that satisfies the end condition and returns a value (runs out of hosts to try)?